README.md

```diff
@@ -5,21 +5,97 @@ Right now this app only works on android. iOS support is planned in the future.
+Select the "SDK Platforms" tab from within the SDK Manager, then check the box next to "Show Package Details" in the bottom right corner. Look for and expand the `Android 14 (UpsideDownCake)` entry, then make sure the following items are checked:
 
+- Android SDK Platform
+- `Intel x86 Atom_64 System Image` or `Google APIs Intel x86 Atom System Image` or (for Apple M1 Silicon) `Google APIs ARM 64 v8a System Image`
+
+You also need to setup your environment variables to use the android utlities just installed. In your `~/.zshrc` or `~/.zprofile` (or `~/.bashrc` file if use bash shell) add the following lines
+
+```sh
+export ANDROID_HOME=$HOME/Library/Android/sdk
+export PATH=$PATH:$ANDROID_HOME/emulator
+export PATH=$PATH:$ANDROID_HOME/platform-tools
+```
+
+then run 
+
+```sh
+source ~/.zshrc
 ```
-sdk.dir=/Users/<USERNAME>/Library/Android/sdk
+
+Verify that `ANDROID_HOME` has been set by running `echo $ANDROID_HOME` and the appropriate directories have been added to your path by running `echo $PATH`.
+
+> Please make sure you use the correct Android SDK path. You can find the actual location of the SDK in the Android Studio "Settings" dialog, under Languages & Frameworks → Android SDK.
+
+### Clone the project
+
+```sh
+git clone https://github.com/Blockchain-Powered-eSIM/LPA.git
 ```
 
-Replace the `<USERNAME>` with your device username
+This project uses React Native's built-in command line interface. Rather than install and manage a specific version of the CLI globally, we recommend you access the current version at runtime using `npx`, which ships with Node.js. With `npx react-native <command>`, the current stable version of the CLI will be downloaded and executed at the time the command is run.
+
+> If you previously installed a global react-native-cli package, please remove it as it may cause unexpected issues:
+> ```sh 
+> npm uninstall -g react-native-cli @react-native-community/cli
+> ```
+
+You will need either an Android device or setup an Android Virtual Device (AVD) to run the app.
+
+If you have a physical android device then you can just use it by connecting to you computer and enable USB debugging. More details are available [here](https://reactnative.dev/docs/running-on-device)
 
-Please run the following command if you run into missing package dependencies during intial build.
+If you plan on using an AVD for development and testing the you'll have to create one using Android Studio Device Manager.
+
+Open `./LPA/android`, you can see the list of available Android Virtual Devices (AVDs) by opening the "AVD Manager" from within Android Studio. If you don't have any virtual device available, create a new one by following the instructions [here](https://developer.android.com/studio/run/managing-avds.html). Preferred image for testing is `UpsideDownCake API Level 34 `. **Make sure that the Android version on your virtual device is greater than Android 10 (or API level 30)**
```

Comment:
To see a list of available AVDs click the "AVD Manager" button in the Android Studio. If the list is empty, create a new AVD by following [this](https://developer.android.com/studio/run/managing-avds.html) link. `UpsideDownCake API Level 34` image is preferred for testing.  
**Make sure that the Android version on the AVD is greater than Android 10 (or API level 30)**